### DISCUSSION TITLE
Performance regression in 5.95.0

### DISCUSSION THREAD

**Discussion by arasmussen** [webpack/webpack#19019] [Q&A]
# Bug report

**What is the current behavior?**

```sh
~/project $ npm install --save webpack@5.95.0
~/project $ rm -rf node_modules/.cache/webpack
~/project $ npx webpack --config config.js
...
webpack 5.95.0 compiled with 2 warnings in 85782 ms
~/project $ npx webpack --config config.js
...
webpack 5.95.0 compiled with 2 warnings in 18755 ms
```

```sh
~/project $ npm install --save webpack@5.94.0
~/project $ rm -rf node_modules/.cache/webpack
~/project $ npx webpack --config config.js
...
webpack 5.94.0 compiled with 2 warnings in 67760 ms
~/project $ npx webpack --config config.js
...
webpack 5.94.0 compiled with 2 warnings in 4278 ms
```

**If the current behavior is a bug, please provide the steps to reproduce.**
I unfortunately don't have repro steps to share... sorry. Mostly posting this as an FYI more than asking for a fix.

I tried disabling a few parts of our config and wasn't able to pinpoint exactly what the issue was. I think we have a pretty standard config file with multiple entries, typescript, babel, scss, images, fonts, etc.

**What is the expected behavior?**
Expected behavior is that overall build time, and particularly cached build time, doesn't regress substantially upon upgrading from 5.94.0 to 5.95.0.

I took a look at the [5.95.0 release notes](https://github.com/webpack/webpack/releases/tag/v5.95.0) and [associated commits](https://github.com/webpack/webpack/compare/v5.94.0...v5.95.0) and nothing really stood out to me.

**Other relevant information:**
webpack version: 5.95.0
Node.js version: 20.15.0
Operating System: macOS 13.6.9
Mode: `production`

Thanks in advance, and totally understand if you want to simply close this one out as I haven't provided repro steps.

**Comment by arasmussen** [comment]
Just noticed https://github.com/webpack/webpack/issues/18914. It's interesting that they're experiencing issues going from 5.95.0 to 5.96.0... I wonder if they were actually on 5.95.0 prior to experiencing perf issues because in my testing the culprit is 5.95.0.

**Comment by arasmussen** [comment]
Just tried running in 5.96.1 as well and perf doesn't seem any different from 5.95.0.

I also tried running in development mode and perf doesn't improve.

**Comment by alexander-akait** [comment]
@arasmussen Please provide the configuration at least, I can't work without code or examples, our benchmark system only signaled this regression - https://github.com/webpack/webpack/discussions/18917#discussioncomment-11155507 (you can check it here - https://webpack.github.io/benchmark/) and it was fixed by disabled `optimization.avoidEntryIife`, some regressions in the `development` mode were not found.

Also you can try - https://webpack.js.org/plugins/profiling-plugin/ to provide us with some data or to see where exactly the regression occurred.

I'll move this to discussions for now until we get more data. But that doesn't mean we're not going to improve or fix it. Waiting for feedback, thank you.
